Rewrote Interaction and Factory creation logic

Daffehh opened the pull request:
### **Refer to commit messages.**
Also, ignore the pull request and branch I closed. Something fucked up happened to the Git integration in my IDE, and it tried to checkout a branch as a new branch 1 million times (or something along those lines).

Commit:
This was mostly just refactoring and cleaning up code that was already
there.

- Remade InteractListener and moved a lot of its code to its own class
called BlockUtil.
- Cleaned up FactoryManager's factory creation.
- Fixed other stuff wrong with code, just look at change summary plz
& ty.

Signed-off-by: Daffehh <simplemodified@gmail.com>

Commit:
I fixed the weirdness that was going on with Factory instances
vs. the FactoryTypes enums and where they should be used. I think?

Signed-off-by: Daffehh <simplemodified@gmail.com>

Commit:
ahhhh test

Signed-off-by: Daffehh <simplemodified@gmail.com>

rxdn commented:
Why are you leaving empty lines at the start and end of functions

ghost commented:
It was the style guide (I know yikes) of the first project I worked on; it kind of stuck to me, sorry. I'll fix it.

nrgill28 committed:
f

rxdn committed:
use transaction scope

nrgill28 committed:
Merge remote-tracking branch 'origin/ryan' into Altii-test

nrgill28 committed:
Factory creation works, removed unnecessary comments

Factories will now be created if they match the recipe for exactly 1 FactoryType. Factories will not be created if one already exists at the same location (to prevent duplicates)
Interacting now uses an empty hand instead of a stick.
Removed long and unnecessary comments from Daffehh, and added one of my own

nrgill28 committed:
Removed a random Minecraft server log file. oops.

nrgill28 committed:
Removed extra whitespace

Commit:
merged altii with current work

Signed-off-by: Daffehh <simplemodified@gmail.com>
